File: VigenereCipher.js
let alpha = 'abcdefghijklmnopqrstuvwxyz';

let cipher_msg = "";
let NEW_KEY = "";
let cipher = ""; 
let encrypt_msg = "";

let searchc = 0;
let searchr = 0;
let encrypt = 0;

//Function to check if message length is too long

function ENCRYPT_MSG(message, key) {
    if(message.length == key.length) {
        return key;
    }
    else
    if(message.length > 5000 || key.length > 5000) {
        console.error("ERROR: string too long.!\n", null);
    }
    else
    NEW_KEY = GEN_NEWKEY(key, message.length, 0);
    return ENCRYPT_MSG = ENCODE(message, NEW_KEY, 0);
}

//Function to generate new key

function GEN_NEWKEY(key, msg_length, k) {
    if(msg_length == 0) {
        return NEW_KEY;
    }
    else
    NEW_KEY = NEW_KEY + key[k%(key.length)];
    return GEN_NEWKEY(key, msg_length-1, k+1);
}

//Function to Encrypt message to Vigenere Cipher

function ENCODE(message, msg_newkey, i) {
    if( i == message.length) {
        return cipher_msg;
    }
    else
    searchc = alpha.indexOf(message[i]);
    searchr = alpha.indexOf(msg_newkey[i]);
    encrypt = (searchc !== -1) && (searchr !== -1) ? (searchr + searchc) : 99;
    cipher = (encrypt >= 26 && encrypt < 52) ? alpha[encrypt - 26] : (encrypt < 26) ? alpha[encrypt] : message[i];
    cipher_msg = cipher_msg + cipher;
    return ENCODE(message, msg_newkey, i+1);
}

console.log(ENCRYPT_MSG("attack at dawn", "secret"));

File: vigenere_cipher.py
alpha = 'abcdefghijklmnopqrstuvwxyz'  # Alphabet used for encryption

def encrypt_msg(message, key):
    """
    Encrypts the message using the Vigenere cipher with the provided key.
    """
    # Check if the message and key are of the same length
    if len(message) == len(key):
        new_key = key
    # Raise an error if the message or key is too long
    elif len(message) > 5000 or len(key) > 5000:
        raise ValueError("ERROR: string too long!")
    # Generate a new key if the lengths are different
    else:
        new_key = gen_new_key(key, len(message))
    
    # Encode the message with the new key
    return encode(message, new_key)

def gen_new_key(key, msg_length):
    """
    Generates a new key by repeating the original key to match the message length.
    """
    new_key = ""
    for i in range(msg_length):
        new_key += key[i % len(key)]
    return new_key

def encode(message, new_key):
    """
    Encodes the message using the Vigenere cipher with the provided key.
    """
    cipher_msg = ""
    for i in range(len(message)):
        searchc = alpha.find(message[i])  # Find the position of the message character in the alphabet
        searchr = alpha.find(new_key[i])  # Find the position of the key character in the alphabet
        if searchc != -1 and searchr != -1:
            encrypt = (searchr + searchc) % 26  # Calculate the encrypted character position
            cipher = alpha[encrypt]  # Get the encrypted character from the alphabet
        else:
            cipher = message[i]  # If character is not in the alphabet, keep it as is
        cipher_msg += cipher  # Append the encrypted character to the cipher message
    return cipher_msg

# Example usage of the encrypt_msg function
print(encrypt_msg("attack at dawn", "lemon"))
